server/src/main/java/org/opensearch/index/translog/TranslogFooter.java

```diff
@@ -0,0 +1,148 @@
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Handles the writing and reading of the translog footer, which contains the checksum of the translog data.
+ *
+ * Each translog file is structured as follows:
+ *
+ * 1. Translog header
+ * 2. Translog operations
+ * 3. Translog footer
+ *
+ * The footer contains the following information:
+ *
+ * - Magic number (int): A constant value that identifies the start of the footer.
+ * - Algorithm ID (int): The identifier of the checksum algorithm used. Currently, this is always 0.
+ * - Checksum (long): The checksum of the entire translog data, calculated using the specified algorithm.
+ */
+public class TranslogFooter {
+
+    /**
+     * FOOTER_LENGTH is the length of the present footer added to translog files.
+     * We write 4 bytes for the magic number, 4 bytes for algorithm ID and 8 bytes for the checksum.
+     * Therefore, the footer length as 16.
+     * */
+    private static final int FOOTER_LENGTH = 16;
+
+    /**
+     * Returns the length of the translog footer in bytes.
+     *
+     * @return the length of the translog footer in bytes
+     */
+    static int footerLength() {
+        return FOOTER_LENGTH;
+    }
+
+    /**
+     * Writes the translog footer to the provided `FileChannel`.
+     *
+     * @param channel  the `FileChannel` to write the footer to
+     * @param checksum the checksum value to be written in the footer
+     * @param toSync   whether to force a sync of the written data to the underlying storage
+     * @return the byte array containing the written footer data
+     * @throws IOException if an I/O error occurs while writing the footer
+     */
+    static byte[] write(FileChannel channel, long checksum, boolean toSync) throws IOException {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        final OutputStreamDataOutput out = new OutputStreamDataOutput(new OutputStreamStreamOutput(byteArrayOutputStream));
+
+        CodecUtil.writeBEInt(out, CodecUtil.FOOTER_MAGIC);
+        CodecUtil.writeBEInt(out, 0);
+        CodecUtil.writeBELong(out, checksum);
+
+        Channels.writeToChannel(byteArrayOutputStream.toByteArray(), channel);
+        if (toSync) {
+            channel.force(false);
+        }
+
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    /**
+     * Reads the checksum value from the footer of the translog file located at the provided `Path`.
+     *
+     * If the translog file is of an older version and does not have a footer, this method returns `null`.
+     *
+     * @param path the `Path` to the translog file
+     * @return the checksum value from the translog footer, or `null` if the footer is not present
+     * @throws IOException if an I/O error occurs while reading the footer
+     */
+    static Long readChecksum(Path path) throws IOException {
```

Comment:
Would prefer non-static functions? Are they getting invoked from a static context?

Reply:
Yes, this method is invoked from static contexts in all instances.
`downloadOnce` in `RemoteFSTranslog` is one instance and `open` in `TranslogReader` is another one.